ocaml/parsing/parser.mly

```diff
@@ -2437,17 +2535,39 @@ class_type_declarations:
 
 /* Core expressions */
 
-seq_expr:
-  | expr        %prec below_SEMI  { $1 }
-  | expr SEMI                     { $1 }
-  | mkexp(expr SEMI seq_expr
+%inline or_function(EXPR):
```

Comment:
Did not review parser changes carefully; assuming that the grammar changes are adapted from upstream.

Reply:
That sounds right to me.